libs/partners/prompty/langchain_prompty/parsers.py

```diff
@@ -1,18 +1,41 @@
 import base64
 import re
-from typing import List, Union
+from typing import Dict, List, Type, Union
 
+from langchain_core.messages import (
+    AIMessage,
+    BaseMessage,
+    FunctionMessage,
+    HumanMessage,
+    SystemMessage,
+)
 from pydantic import BaseModel
 
 from .core import Invoker, Prompty, SimpleModel
 
 
+class RoleMap:
```

Comment:
Does this cover the full scope of roles? What about `"tool"`?

Reply:
Yes, it's on par with the latest versions of Prompty, they define the [same set of roles](https://github.com/microsoft/prompty/blob/074abe7f35968e9ed10a54eeda98d203c00de3ae/runtime/prompty/prompty/parsers.py#L11)! Tools are defined in the [YAML header](https://github.com/microsoft/prompty/blob/074abe7f35968e9ed10a54eeda98d203c00de3ae/web/docs/prompty-file-spec/page.mdx#parameters-for-the-model-attribute) of the `.prompty` files, [here's an example](https://github.com/microsoft/prompty/blob/074abe7f35968e9ed10a54eeda98d203c00de3ae/runtime/prompty/tests/prompts/funcfile.prompty#L11)